pkg/build/controller/config_controller.go

```diff
@@ -6,19 +6,38 @@ import (
 	"github.com/golang/glog"
 	kapi "k8s.io/kubernetes/pkg/api"
 	kerrors "k8s.io/kubernetes/pkg/api/errors"
-	"k8s.io/kubernetes/pkg/util"
+	kutil "k8s.io/kubernetes/pkg/util"
 
 	buildapi "github.com/openshift/origin/pkg/build/api"
 	buildclient "github.com/openshift/origin/pkg/build/client"
+	"github.com/openshift/origin/pkg/build/util"
 )
 
 type BuildConfigController struct {
 	BuildConfigInstantiator buildclient.BuildConfigInstantiator
+	BuildConfigDeleter      buildclient.BuildConfigDeleter
+	BuildDeleter            buildclient.BuildDeleter
+	BuildLister             buildclient.BuildLister
 }
 
 func (c *BuildConfigController) HandleBuildConfig(bc *buildapi.BuildConfig) error {
 	glog.V(4).Infof("Handling BuildConfig %s/%s", bc.Namespace, bc.Name)
 
+	if !bc.DeletionTimestamp.IsZero() {
+		builds, err := c.BuildLister.List(bc.Namespace, util.ConfigSelector(bc.Name), nil)
+		if err != nil {
+			return err
+		}
+
+		err = nil
+		for _, build := range builds.Items {
+			if err = c.BuildDeleter.Delete(build.Namespace, build.Name); err != nil {
+				glog.Errorf("Cannot delete Build %s/%s: %v", build.Namespace, build.Name, err)
+			}
+		}
+		return err
```

Comment:
is this error type going to be handled as a Retryable error or a fatal error by the RetryController that wraps the BuildConfigController?  I'm guessing we do want to retry this.

Also I don't love that you're just returning the last error, it might be better to create a custom error type that embeds an array of errors for all the failed deletions, particularly since they might have different reasons.